internal/ingress/backend/endpoint.go

```diff
@@ -54,6 +56,50 @@ func (resolver *endpointResolver) Resolve(ingress *extensions.Ingress, backend *
 	return resolver.resolveIP(ingress, backend)
 }
 
+// For each item in the targets slice, returns the corresponding pod in the result slice at the same index. The result slice is exactly as long as the input slice.
+func (resolver *endpointResolver) ReverseResolve(ingress *extensions.Ingress, backend *extensions.IngressBackend, targets []*elbv2.TargetDescription) ([]*corev1.Pod, error) {
+	service, servicePort, err := findServiceAndPort(resolver.store, ingress.Namespace, backend.ServiceName, backend.ServicePort)
+	if err != nil {
+		return nil, err
+	}
+	serviceKey := ingress.Namespace + "/" + service.Name
+	eps, err := resolver.store.GetServiceEndpoints(serviceKey)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to find service endpoints for %s: %v", serviceKey, err.Error())
+	}
+
+	targetsMap := map[string]int{}
+	for i, target := range targets {
+		targetsMap[aws.StringValue(target.Id)] = i
+	}
+
+	result := make([]*corev1.Pod, len(targets))
+	for _, epSubset := range eps.Subsets {
+		for _, epPort := range epSubset.Ports {
+			// servicePort.Name is optional if there is only one port
+			if servicePort.Name != "" && servicePort.Name != epPort.Name {
+				continue
+			}
+			for _, epAddr := range append(epSubset.Addresses, epSubset.NotReadyAddresses...) {
+				if epAddr.TargetRef == nil || epAddr.TargetRef.Kind != "Pod" {
+					continue
+				}
+
+				podKey := ingress.Namespace + "/" + epAddr.TargetRef.Name
+				pod, err := resolver.store.GetPod(podKey)
+				if err != nil {
+					continue
+				}
+
```

Comment:
@M00nF1sh @devkid  shouldn't we also check if DeletionTimestamp is != nil? because pods, that are in state terminating can still be found in the store, but should already be drained, right?

Reply:
Pods in `Terminating` state appear neither in `Addresses` nor in `NotReadyAddresses`.

Reply:
:man_shrugging: for us they do, that's why I mentioned it here.

Reply:
That's weird. I actually thought about implementing the check for `DeletionTimestamp != nil` but I verified on our end that terminating pods do *not* appear in `NotReadyAddresses`. Feel free to add the check. I'll have another look at this on our end as well next week.